oap-server/server-receiver-plugin/configuration-discovery-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/recevier/configuration/discovery/ConfigurationDiscoveryProvider.java

```diff
@@ -0,0 +1,98 @@
+import org.apache.skywalking.oap.server.library.module.ServiceNotProvidedException;
+import org.apache.skywalking.oap.server.library.util.ResourceUtils;
+import org.apache.skywalking.oap.server.receiver.sharing.server.SharingServerModule;
+import org.apache.skywalking.oap.server.recevier.configuration.discovery.handler.grpc.ConfigurationDiscoveryServiceHandler;
+
+public class ConfigurationDiscoveryProvider extends ModuleProvider {
+
+    private ConfigurationDiscoveryRulesWatcher configurationDiscoveryRulesWatcher;
+
+    @Override
+    public String name() {
+        return "default";
+    }
+
+    @Override
+    public Class<? extends ModuleDefine> module() {
+        return ConfigurationDiscoveryModule.class;
+    }
+
+    @Override
+    public ModuleConfig createConfigBeanIfAbsent() {
+        return new EmptySettings();
+    }
+
+    @Override
+    public void prepare() throws ServiceNotProvidedException, ModuleStartException {
+        Reader applicationReader;
+        try {
+            applicationReader = ResourceUtils.read("agent-dynamic-config.yml");
+        } catch (FileNotFoundException e) {
+            throw new ModuleStartException("can't load agent-dynamic-config.yml", e);
+        }
+        ConfigurationDiscoveryRulesReader reader = new ConfigurationDiscoveryRulesReader(applicationReader);
+        ConfigurationDiscoveryRules configurationDiscoveryRules = reader.readRules();
+
+        configurationDiscoveryRulesWatcher = new ConfigurationDiscoveryRulesWatcher(configurationDiscoveryRules, this);
+        this.registerServiceImplementation(
+            ConfigurationDiscoveryService.class,
+            new DefaultConfigurationDiscoveryService(configurationDiscoveryRulesWatcher)
+        );
+    }
+
+    @Override
+    public void start() throws ServiceNotProvidedException, ModuleStartException {
+        DynamicConfigurationService dynamicConfigurationService = getManager().find(ConfigurationModule.NAME)
+                                                                              .provider()
+                                                                              .getService(
+                                                                                  DynamicConfigurationService.class);
+        dynamicConfigurationService.registerConfigChangeWatcher(configurationDiscoveryRulesWatcher);
+
+        GRPCHandlerRegister grpcHandlerRegister = getManager().find(SharingServerModule.NAME)
+                                                              .provider()
+                                                              .getService(GRPCHandlerRegister.class);
+        grpcHandlerRegister.addHandler(new ConfigurationDiscoveryServiceHandler(getManager()));
+    }
+
+    @Override
+    public void notifyAfterCompleted() throws ServiceNotProvidedException, ModuleStartException {
+    }
+
+    @Override
+    public String[] requiredModules() {
+        return new String[] {
+            CoreModule.NAME,
```

Comment:
What do you need `Core` for?

Reply:
CoreModule is not needed, I will modify it.